scripts/source-maintenance.sh

```diff
@@ -61,9 +70,15 @@ else
 	MD5="md5sum"
 fi
 
-ASVER=`astyle --version 2>&1 | grep -o -E "[0-9.]+"`
-if test "${ASVER}" != "2.04" ; then
-	echo "Astyle version problem. You have ${ASVER} instead of 2.04"
+${ASTYLE} --version >/dev/null 2>/dev/null
+result=$?
+if test $result -gt 0 ; then
+	echo "ERROR: cannot run ${ASTYLE}"
+	exit 1
+fi
+ASVER=`${ASTYLE} --version 2>&1 | grep -o -E "[0-9.]+"`
+if test "${ASTYLE}" = "astyle" &&  test "${ASVER}" != "${TargetAstyleVersion}" ; then
+	echo "Astyle version problem. You have ${ASVER} instead of ${TargetAstyleVersion}"
 	ASVER=""
```

Comment:
AFAICT, this code is still missing the point of why the warning or error is needed and what should happen when there is a version mismatch:

1. If the astyle version is not the officially used one, we should tell the user. This rule remains active _regardless_ of whether astyle binary or version has been manually configured. We should warn the user that there may be trouble because it is trivial for the user to make a mistake here and not know it, even if the user knows what they are doing and are specifying an explicit binary location.

2. If the user has not specified an astyle location, then the version mismatch may disable formatting. If we decide to disable formatting, then we should tell the user that they are not getting something they probably want and expect -- not just that they are not using an official version, but that their sources are not going to be formatted. Without an explicit warning about skipped formatting, that outcome would be very obscure to all but the true experts. Why continue without formatting in this case? This is a so called "safe" default, at least for those of us who believe that the user runs this script for reasons other than wanting to format the sources and does not really care much if some of the formatting disappears -- it is still considered a "successful" outcome. FWIW, I do not share that belief, but at least the user will be warned.

3. If the user has specified astyle location, in any shape or form, then the version mismatch must not disable formatting. We should proceed with using the binary specified by the user. The warning in item 1 is necessary and sufficient in this case. We should trust that the user knows what they are (explicitly) doing, especially since malformed sources are now easy to revert with `git`.

Reply:
Done